src/main.ts

```diff
@@ -0,0 +1,75 @@
+import * as core from '@actions/core'
+import * as github from '@actions/github'
+
+const oktokit = github.getOctokit(core.getInput('token'))
+
+async function run() {
+  const owner = github.context.repo.owner
+  const repo = github.context.repo.repo
+  const upstream_owner = core.getInput('upstream_owner')
+  const upstream_repo = core.getInput('upstream_repo')
+  const upstream_branch =
+    core.getInput('upstream_branch') ||
+    (await oktokit.rest.repos.get({owner: upstream_owner, repo: upstream_repo}))
+      .data.default_branch
+  const reviewers = core.getInput('reviewers').split(',')
+  const base_branch = (await oktokit.rest.repos.get({owner, repo})).data
+    .default_branch
+
+  try {
+    const {data: autosync_pulls} =
+      await oktokit.rest.search.issuesAndPullRequests({
+        q: 'label:autosync+state:open+is:pull-request',
+        per_page: 1
+      })
+    if (autosync_pulls.total_count < 1) {
+      const {data: upstream_manifest} = await oktokit.rest.repos.getContent({
+        owner: upstream_owner,
+        repo: upstream_repo,
+        ref: upstream_branch,
+        path: 'src/manifest.json'
+      })
+      const {data: base_manifest} = await oktokit.rest.repos.getContent({
+        owner,
+        repo,
+        path: 'src/manifest.json'
+      })
+      if ('content' in base_manifest && 'content' in upstream_manifest) {
```

Comment:
what means the other case? I mean the `else` -- would it be an unexpected response of oktokit? Is logging more appropriate or `core.setFailed(error)`?

Reply:
See https://github.com/GOG-Nebula/action-sync-fork/pull/1#issuecomment-1146227922